healsparse/bitSparseMap.py

```diff
@@ -0,0 +1,310 @@
+                        return self._clear_bits_at_locs(indices)
+
+                elif isinstance(value, np.ndarray):
+                    if value.dtype != self._dtype:
+                        raise ValueError("Can only set to array of bools")
+                    if len(value) != len(indices):
+                        raise ValueError("Length mismatch")
+                    self._set_bits_at_locs(indices[value])
+                    self._clear_bits_at_locs(indices[~value])
+                else:
+                    raise ValueError("Can only set to bool or array of bools")
+        elif isinstance(key, np.ndarray):
+            if not is_integer_value(key.dtype.type(0)):
+                raise IndexError("Numpy array indices must be integers for __setitem__")
+            if isinstance(value, (bool, np.bool_)):
+                if value:
+                    return self._set_bits_at_locs(key)
+                else:
+                    return self._clear_bits(self._data, key)
+            elif isinstance(value, np.ndarray):
+                if value.dtype != self._dtype:
+                    raise ValueError("Can only set to bool or array of bools")
+                if len(value) != len(key):
+                    raise ValueError("Length mismatch")
+                self._set_bits_at_locs(key[value])
+                self._clear_bits_at_locs(key[~value])
+            else:
+                raise ValueError("Can only set to bool or array of bools")
+        elif isinstance(key, (list, tuple)):
+            indices = np.atleast_1d(key)
+            if len(indices) > 0:
+                if not is_integer_value(indices[0]):
+                    raise IndexError("List array indices must be integers for __setitem__")
+            if isinstance(value, (bool, np.bool_)):
+                if value:
+                    return self._set_bits_at_locs(indices)
+                else:
+                    return self._clear_bits_at_locs(indices)
+            elif isinstance(value, np.ndarray):
+                if value.dtype != self._dtype:
+                    raise ValueError("Can only set to bool or array of bools")
+                if len(value) != len(key):
+                    raise ValueError("Length mismatch")
+                self._set_bits_at_locs(indices[value])
+                self._clear_bits_at_locs(indices[~value])
+            else:
+                raise ValueError("Can only set to bool or array of bools")
+        else:
+            raise IndexError("Illegal index type (%s) for __setitem__ in BitSparseMap." %
+                             (key.__class__))
+
+    def __array__(self):
+        return np.unpackbits(self._data, bitorder="little")
+
+    def __and__(self, other):
+        pass
+
+    def __iand__(self, other):
+        pass
+
+    def __or__(self, other):
+        pass
+
+    def __ior__(self, other):
+        pass
+
+    def __xor__(self, other):
+        pass
+
+    def __ixor__(self, other):
+        pass
```

Comment:
As far as I can tell, these methods will fail silently if anyone tried to use them. I'd rather they raise a NotImplementedError or we add tests showing they work.